src/js/jsx/search/SearchBar.jsx

```diff
@@ -88,50 +91,188 @@ define(function (require, exports, module) {
+                ancestors = layerTree.ancestors(layer),
+                ancestorNames = ancestors.first().name;
+
+            ancestors.skip(1).forEach(function (ancestor) {
+                ancestorNames += ("/" + ancestor.name);
+            });
+            return ancestorNames;
+        },
+
+        /**
+         * Make list of layers in the current document to be used as dropdown options
+         * 
          * @return {Array.<Object>}
          */
-        _getSelectOptions: function () {
-            // get list of layers
+        _getLayerOptions: function () {
+            // Get list of layers
             var appStore = this.getFlux().store("application"),
                 document = appStore.getCurrentDocument(),
                 layers = document.layers.allVisible.reverse(),
                 layerMap = layers.map(function (layer) {
                     // Used to determine the layer face icon
-                    var iconID = "layer-";
-                    if (layer.isArtboard) {
-                        iconID += "artboard";
-                    } else if (layer.kind === layer.layerKinds.BACKGROUND) {
-                        iconID += layer.layerKinds.PIXEL;
-                    } else if (layer.kind === layer.layerKinds.SMARTOBJECT && layer.isLinked) {
-                        iconID += layer.kind + "-linked";
-                    } else {
-                        iconID += layer.kind;
-                    }
-                    
-                    return { id: "layer_" + layer.id.toString(), title: layer.name, type: "item", svgType: iconID };
-                }),
-                layerLabel = { id: "layer_header", title: "Layers", type: "header" },
+                    var iconID = this._getSVGInfo(layer),
+                        ancestry = this._formatLayerAncestry(layer);
+
+                    return {
+                        id: "layer_" + layer.id.toString(),
+                        title: layer.name,
+                        info: ancestry,
+                        displayInfo: ancestry,
+                        svgType: iconID,
+                        type: "item"
+                    };
+                }.bind(this)),
+
+                // Get shortest unique layer ancestry
+                ancestors = collection.pluck(layerMap, "info"),
+                shortenedPaths = pathUtil.getShortestUniquePaths(ancestors.toArray());
+
+            layerMap.map(function (layerItem, index) {
+                var newInfo = shortenedPaths[index];
+                
+                // Only show ancestry if other layers have the same name
+                if (layerItem.title === newInfo) {
+                    layerItem.displayInfo = "";
+                } else {
+                    layerItem.displayInfo = newInfo;
+                }
+            });
+
+            var layerLabel = {
+                    id: "layer_header",
+                    title: "Layers",
```

Comment:
Sorry for the jargon, what I meant there was that the string "Layers" eventually needs to be specified in a way that will make it possible for it to be translated to other languages (i.e., for use with other locales). We do that in two parts 1) user-visible strings are added to a dictionary here https://github.com/adobe-photoshop/spaces-design/blob/955e1995d80c81c0c8591f3dcbb8146303f7b219/src/nls/root/strings.js#L354 and 2) we refer to those strings using keys into the dictionary, as in here: https://github.com/adobe-photoshop/spaces-design/blob/955e1995d80c81c0c8591f3dcbb8146303f7b219/src/js/jsx/sections/pages/PagesPanel.jsx#L493